Ewout,

would you mind rewriting your PR (for the moment) as
- gcc + clang
- O0 + O2
- C11 only


About clang error
```
src/stx.c:40:18: error: expression is not an integer constant expression
    TYPE1 = (int)log2(sizeof(Head1)), 
            ~~~~~^~~~~~~~~~~~~~~~~~~
```
I'm puzzled... If you know how to solve. it please add a patch to stx.c

Also, feel free to suggest improvements to the Makefile (I'm not sure it's state-of-the art)

++!

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -7,18 +7,10 @@ jobs:
       fail-fast: false
       matrix:
         CC: [gcc, clang]
-        STD: [c99, c11, c17]
-        OPTIM: [-O0]
-        include:
-          - CC: gcc
-            STD: c17
-            OPTIM: -O3
-          - CC: clang
-            STD: c17
-            OPTIM: -O3
+        OPTIM: [-O0, -O2]
 
     env:
-      make_options: 'CC=${{ matrix.CC }} STD=${{ matrix.STD }} OPTIM=${{ matrix.OPTIM }}' 
+      make_options: 'CC=${{ matrix.CC }} OPTIM=${{ matrix.OPTIM }}'
 
     steps:
     - uses: actions/checkout@v2

